documentation/source/best_practices.rst

```diff
@@ -0,0 +1,50 @@
+.. _best_practices:
+
+**************
+Best Practices
+**************
+
+.. _performance:
+
+Performance Considerations
+==========================
+
+In order to get the maximum performance out of a cocotb testbench,
+the following points may help you.
+
+
+Clock Generator in HDL
+----------------------
+
+Generate your clock with a HDL module instead of using :class:`~cocotb.clock.Clock`.
+Doing so prevents calls back into Python every single clock half-period.
+
+Performance Impact: **Major**
+
+
+Reusing Objects
+---------------
+
+Do not create single-use objects in a loop but instead create them outside the loop and reuse them.
+Typical cases are :class:`~cocotb.triggers.Timer` or :class:`~cocotb.clock.Clock` objects as shown below:
```

Comment:
```suggestion
Typical cases are :class:`~cocotb.triggers.Timer` or :class:`~cocotb.triggers.RisingEdge` of the clock signal, as shown below:
```

You shouldn't create multiple `cocotb.clock.Clock`s at all, but `RisingEdge(clk)` is very common.

Reply:
Indeed, changed.